packages/app/components/settings/settings-wallet-item.tsx

```diff
@@ -0,0 +1,209 @@
+import React, { useMemo } from "react";
+
+import { SvgProps } from "react-native-svg";
+
+import { useIsDarkMode } from "@showtime-xyz/universal.hooks";
+import {
+  Ethereum,
+  Tezos,
+  Check,
+  Mail,
+  PhonePortraitOutline,
+  GoogleOriginal,
+  Apple,
+} from "@showtime-xyz/universal.icon";
+import { PressableHover } from "@showtime-xyz/universal.pressable-hover";
+import { colors } from "@showtime-xyz/universal.tailwind";
+import { Text } from "@showtime-xyz/universal.text";
+import { View } from "@showtime-xyz/universal.view";
+
+import { useSetPrimaryWallet } from "app/hooks/api/use-set-primary-wallet";
+import { useCurrentUserAddress } from "app/hooks/use-current-user-address";
+import { useUser } from "app/hooks/use-user";
+import { WalletAddressesV2 } from "app/types";
+import { formatAddressShort } from "app/utilities";
+
+import { Hidden } from "design-system/hidden";
+
+import { WalletDropdownMenu } from "./wallet-dropdown-menu";
+
+type Props = {
+  wallet: WalletAddressesV2;
+  onEditNickname: (item: WalletAddressesV2) => void;
+};
+
+type WalletLoginWithProps = {
+  Icon?: (props: SvgProps) => JSX.Element;
+  text: string;
+};
+const LoginWithPlatformTitle = ({ Icon, text }: WalletLoginWithProps) => {
+  const isDark = useIsDarkMode();
+
+  return (
+    <View tw="flex-row items-center">
+      {Icon && (
+        <Icon
+          color={isDark ? colors.white : colors.gray[900]}
+          width={16}
+          height={16}
+        />
+      )}
+      <Text tw="ml-1 text-sm font-medium text-gray-900 dark:text-white">
+        {text}
+      </Text>
+    </View>
+  );
+};
+
+const LoginWithTitle = ({
+  wallet,
+  nickname,
+}: {
+  wallet: WalletAddressesV2;
+  nickname?: string;
+}) => {
+  const Title = useMemo(() => {
+    if (wallet.email && wallet.is_email) {
+      return <LoginWithPlatformTitle text={wallet.email} Icon={Mail} />;
+    }
+    if (wallet.phone_number && wallet.is_phone) {
+      return (
+        <LoginWithPlatformTitle
+          text={wallet.phone_number}
+          Icon={PhonePortraitOutline}
+        />
+      );
+    }
+    if (wallet.is_google) {
+      return (
+        <LoginWithPlatformTitle
+          text={"Login from Google"}
+          Icon={GoogleOriginal}
+        />
+      );
+    }
+    if (wallet.is_apple) {
+      return <LoginWithPlatformTitle text={"Login from Apple"} Icon={Apple} />;
+    }
+    return null;
+  }, [
+    wallet.email,
+    wallet.is_apple,
+    wallet.is_email,
+    wallet.is_google,
+    wallet.is_phone,
+    wallet.phone_number,
+  ]);
```

Comment:
Defining components like this is considered anti pattern. I actually think there is no point of memo, it should be enough to just return based on if condition.

Reply:
yeah! just moved it!